.github/workflows/rust.yml

```diff
@@ -8,14 +8,32 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  check_n_test:
-    name: cargo check & test
-    uses: noir-lang/.github/.github/workflows/rust-test.yml@main
+  nix:
```

Comment:
Are we fine with **only** Nix testing or do we also want a globally installed barretenberg test? Like https://github.com/noir-lang/barretenberg-sys/blob/279be62243b8c51159b75960d536760d69d83990/.github/workflows/test.yml#L45-L95

Reply:
Sounds like everyone is fine with this change. This will diverge from the reusable workflows until we get nix throughout all repos.